architecture.svg

```diff
@@ -0,0 +1 @@
AAA9FU2xpY2VWZXJ0QWxpZ24AAAAHZGVmYXVsdAAAAAtiZ0NvbG9yVHlwZWVudW0AAAARRVNsaWNlQkdDb2xvclR5cGUAAAAATm9uZQAAAAl0b3BPdXRzZXRsb25nAAAAAAAAAApsZWZ0T3V0c2V0bG9uZwAAAAAAAAAMYm90dG9tT3V0c2V0bG9uZwAAAAAAAAALcmlnaHRPdXRzZXRsb25nAAAAAAA4QklNBCgAAAAAAAwAAAACP/AAAAAAAAA4QklNBBQAAAAAAAQAAAACOEJJTQQMAAAAAAgWAAAAAQAAAKAAAAA4AAAB4AAAaQAAAAf6ABgAAf/Y/+0ADEFkb2JlX0NNAAH/7gAOQWRvYmUAZIAAAAAB/9sAhAAMCAgICQgMCQkMEQsKCxEVDwwMDxUYExMVExMYEQwMDAwMDBEMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMAQ0LCw0ODRAODhAUDg4OFBQODg4OFBEMDAwMDBERDAwMDAwMEQwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAz/wAARCAA4AKADASIAAhEBAxEB/90ABAAK/8QBPwAAAQUBAQEBAQEAAAAAAAAAAwABAgQFBgcICQoLAQABBQEBAQEBAQAAAAAAAAABAAIDBAUGBwgJCgsQAAEEAQMCBAIFBwYIBQMMMwEAAhEDBCESMQVBUWETInGBMgYUkaGxQiMkFVLBYjM0coLRQwclklPw4fFjczUWorKDJkSTVGRFwqN0NhfSVeJl8rOEw9N14/NGJ5SkhbSVxNTk9KW1xdXl9VZmdoaWprbG1ub2N0dXZ3eHl6e3x9fn9xEAAgIBAgQEAwQFBgcHBgU1AQACEQMhMRIEQVFhcSITBTKBkRShsUIjwVLR8DMkYuFygpJDUxVjczTxJQYWorKDByY1wtJEk1SjF2RFVTZ0ZeLys4TD03Xj80aUpIW0lcTU5PSltcXV5fVWZnaGlqa2xtbm9ic3R1dnd4eXp7fH/9oADAMBAAIRAxEAPwD1VJJJJSkkkklKSSSSUpJJJJSkkkklKSSSSUpJJJJSkkkklKSSSSUpJJJJT//Q9VSSSSU8v9ZfrP1HpHU6aKaWOxvTbbZuBLrBuc21lTmua2t1bG/uWe966RuRQ7HGSHj0Cz1BZ+bsI37/AOrtWJ9c+mfbOknJrE34JNrY5NcfrDP+2/0v/WVzLevuH1Ud0kO/Tmz0AdZ+zOHqud/V/wC0f9RJTbyPr11W259mDRUzEZqBYx73bCdjLL3sexlPqO/9J77EW/68dTyWNZ03CixjN+SS19+397YyjZ+h/wCGu/7aQaem/ZfqRmZlgi7PNVmvaoWVjHb/AGmudd/15aP+L8D7DmOgbjkAE94FdcD/AKSKmlj/AF+zfsz23YrLspxAx31ktrM6fpK5ts3fuel/Pf8AAqNX146vi5W3qeMz0mketU2t9VrGnXc1tz37vb7tj2/pP9IqvRGt/wCebWwNrcrL2iNBAyNsf1VY/wAYIH2/GManHeCfg7T/AKpJT0eN1u2z6w3dKuYxtDqGZGDa0mbGkD1N0+13u3+ns/0Kb60dct6NhV2YzGWZVz9lbLJ2wAX2vOyHfRbs/wCuLK6204OJ0Pr9Y/oIqryI5NNjWtd/36tv/Hq0WN6t9b7A8b8TpGPsiZabslvv/wDZf2/2EFO9h5VeZiU5dX83extjfg4bljdP6/1PPw+oZmJhsyW0X+jgVtf6ZtaC1r7LH2+z2td6v/gSx8bqlnSPq11Tpj3E5XTbXYtHYlt5P2exv+dda3/i1oPvyfq90rpfRcCplnUsv2NNh/RtdpZk22bPc7a+32/yP+L9OxKSPzfrvVX678DDtY3V1Fdj/Vjwa4+zd/VVgfWRmR9W8jrOGyLaWOmmzhtrR9B+2Nzfdv8A5df+jTN6Z9agA89arL+TWcRhZP7u4WNt2rDwxYPq39ZG2Fpsbk3+oWCGF8M9Q1t/Nr3/AEElN9v1n6xn1109EwWZOSyqt+Xc922hlrmh78dvuZvcxzv9J/1uxW8zr2e/OPSuj4zMrNqaHZdr3EY9JP5jnwH2u/q/+lfStfVjGrx+g4QrEerU2558X2j1nk/565/6t/t+eonp7cM2HMt+1HKNgt9Sf+Ba79F+5/L9VFTpjr3Vum5VNH1gxqa8fJcK6s3Gc41NeeGXNt97P6//AKl9PSxs3LtzrKLKg2pkwYIIg6btf/Vn+B9ldix+rdN+tnVsC3ByW9OFdsHcx124FpD2lu6p37q6LGZbXj1V3O32sY1tjx3cBDnf2nIKSpJJJKf/0fVUkkklLEAiDqDyFxLvqFlHqBrbbWOll8zLvVFUz9nFe3bv/wAD6vq/Q/S/8Gu3SSU5/W+mv6h0i/AoLanva0VSPaCxzbGNO36LPZsVX6q9Gyuk4NteW5huutNhbWS5rRtZW1u9zWbnez91bSSSnlunfVbOxfrK/qVllZxW2XW1wSbHetvitzNrWM9P1fp+p/5gT62fVvP6vdj3YT6wWMdVY20lsBxa5tjdjLN2385i6VJJTQyul15PRn9Ke6WuoFIeRw5rYrt2/wAl7W2Kt9WOjXdIwHV5LmPyrrDZc6slzeAytjXvax7ttbP3VsJJKed6j9WH5f1jx+ptcwYo2PyqyTudZTPoFrdux279E1+//R/8IrnX+iW9TGPkYl/2bPwnmzGtI3NkxuZY3912xi1kklOCKvrlkj0L7cLDrOlmTjiyy2O/oV3baq3u/fs9TZ/o0DF+rGXjdF6t01llc5llhxCXOMMLWMp+0P2bvV9n6XY2xdKkkprdMxrMTpuLiWEOsx6a6nls7SWNax22fzfasrO6F1CnqL+q9CyK8fIvgZWNcCaLY+i93p++uz+r/wCB77vW3kklPOu6b9aupPYzqWZTg4rSHPZ08vFr4O7ab7PdU3/i10SSSSlJJJJKf//S9VSXyqkkp+qkl8qpJKfqpJfKqSSn6qSXyqkkp+qkl8q
```

Comment:
Should `Machine CRs` be `Machine CRDs`?

It might also be valuable to show the common controller-manager and mention MachineSets, MachineDeployments, and MachineClasses somehow.

Reply:
I think it's right as is, no? There is one instance of the CRD _definition_ and multiple instances of the CR.

Reply:
Ah, I missed that distinction the first time through

Reply:
loving the email notifications i'm receiving from this code review thread. :laughing: 
![cluster-api-reviews](https://user-images.githubusercontent.com/331852/53672610-f7619900-3c8b-11e9-938e-084b046da3bd.png)

Reply:
I agree with @detiber that showing Cluster, MachineSet ... would be valuable to represent the extent beyond cluster/machines. Not sure if making the explicit distinction between CRD and CR at this level is required or can be omitted to simplify the drawing. 

Also, I'm not sure the "update state" arrow from cluster A makes sense given that the resources have been pivoted. Are there still some updates? Or should that arrow come from cluster B.